src/sentry/snuba/tasks.py

```diff
@@ -114,7 +115,8 @@ def delete_subscription_from_snuba(query_subscription_id, **kwargs):
             QueryDatasets(subscription.snuba_query.dataset), subscription.subscription_id
         )
 
-    subscription.delete()
+    if subscription.status == QuerySubscription.Status.DELETING.value:
+        subscription.delete()
```

Comment:
I guess we need to set the `subscription_id` to `None` here if the status is `DISABLING`